cmd/gardenlet/app/migration.go

```diff
@@ -149,3 +162,47 @@ func createOSCHashMigrationSecret(ctx context.Context, seedClient client.Client)
 	}
 	return flow.Parallel(tasks...)(ctx)
 }
+
+// TODO (shafeeqes): Remove this function after gardener v1.99 has been released
+func cleanupDWDAccess(ctx context.Context, gardenClient client.Client, seedClient client.Client) error {
+	shootList := &gardencorev1beta1.ShootList{}
+	if err := gardenClient.List(ctx, shootList); err != nil {
```

Comment:
Can't we only select the shoots for which this gardenlet is responsible? The `shoot.spec.seedName` can be used as a field selector for the `List` call. Then I guess you don't need to also retrieve all shoot control plane namespaces in the seed.